spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/stream/SkipperStreamDeployer.java

```diff
@@ -263,20 +264,40 @@ private String determinePlatformName(final String platformName) {
 	}
 
 	private void validateAllAppsRegistered(StreamDeploymentRequest streamDeploymentRequest) {
+		StreamDefinition streamDefinition = this.streamDefinitionRepository.findOne(streamDeploymentRequest.getStreamName());
 		for (AppDeploymentRequest adr : streamDeploymentRequest.getAppDeploymentRequests()) {
 			String version = ResourceUtils.getResourceVersion(adr.getResource());
-			validateAppVersionIsRegistered(adr, version);
+			validateAppVersionIsRegistered(getRegisteredName(streamDefinition, adr.getDefinition().getName()), adr, version);
 		}
 	}
 
-	public void validateAppVersionIsRegistered(AppDeploymentRequest appDeploymentRequest, String appVersion) {
-		String name = appDeploymentRequest.getDefinition().getName();
+	private String getRegisteredName(StreamDefinition streamDefinition, String adrAppName) {
+		for (StreamAppDefinition appDefinition: streamDefinition.getAppDefinitions()) {
+			if (appDefinition.getName().equals(adrAppName)) {
+				return appDefinition.getRegisteredAppName();
+			}
+		}
+		return adrAppName;
+	}
+
+	public void validateAppVersionIsRegistered(StreamDefinition streamDefinition, AppDeploymentRequest appDeploymentRequest, String appVersion) {
```

Comment:
Except the first line this method is identical to https://github.com/spring-cloud/spring-cloud-dataflow/pull/2351/files#diff-50f573b367c4c71c1abdc05c961b88f9R294 and can use it internally